repo-projects/Directory.Build.props

```diff
@@ -77,7 +77,9 @@
     <EnvironmentVariables Include="GIT_COMMIT=$(GitCommitHash)" Condition="'$(GitCommitHash)' != ''" />
     <EnvironmentVariables Include="RepositoryType=Git" />
 
-    <EnvironmentVariables Include="DeterministicSourcePaths=false" />
+    <EnvironmentVariables Include="DeterministicSourcePaths=true" Condition="'$(DeterministicBuildOptOut)' != 'true'" />
+    <EnvironmentVariables Include="DeterministicSourcePaths=false" Condition="'$(DeterministicBuildOptOut)' == 'true'" />
+
```

Comment:
As mentioned, we shouldn't need this anymore:

```suggestion
```

Reply:
I understand [your comment](https://github.com/dotnet/source-build-externals/pull/281#issuecomment-2044596212) to mean that we'd like `DeterministicSourcePaths=true` to be applied also when this repo is built stand-alone.

Did I misunderstand? Or is there something else that will set it in the stand-alone case?

Reply:
I was suggesting to enable deterministic source paths in both the VMR and the SBE repo build unless there's a technical reason to not enable that.

Reply:
`DeterministicSourcePaths` already defaults to true when running on CI:
- https://github.com/dotnet/roslyn/blob/d60cfabb92b1269ad24e7bc303face107eaecf00/src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets#L208
- https://github.com/dotnet/sdk/blob/6a719af0884316c3482aaedcb755afb9d339a416/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props#L43

In a perfect world we shouldn't need any of these env vars. Of course this depends on whether repos don't change the default but I would be surprised if they do. The CI binlogs from the VMR should tell.

Reply:
> DeterministicSourcePaths already defaults to true when running on CI:

The lines are needed to enable DeterministicSourcePaths when the build is not running on CI (like with the vmr).

Do we want that, or not?

Reply:
When performing a local build of the VMR, you will eventually get dev versions, i.e. `9.0.0-dev`. That's tracked as part of https://github.com/dotnet/source-build/issues/3934. My expectation is that for producing the shipping artifacts with the correct versions, the `--ci` flag an probably an additional parameter needs to be passed in. TBD as far as I know. cc @mmitche

Reply:
I've removed the lines since no other repos are handling this, `source-build-externals` should also defer it to the vmr.